resources/xsd/fastRTPS_profiles.xsd

```diff
@@ -5,293 +5,1124 @@
+            <xs:element name="durability" type="durabilityQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="latencyBudget" type="latencyBudgetQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="lifespan" type="lifespanQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="liveliness" type="livelinessQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="ownership" type="ownershipQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="ownershipStrength" type="ownershipStrengthQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="partition" type="partitionQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="publishMode" type="publishModeQosPolicyType" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="reliability" type="reliabilityQosPolicyType" minOccurs="0" maxOccurs="1"/>
         </xs:all>
     </xs:complexType>
 
-    <xs:simpleType name="ipv4AddressType">
+    <!--History memory Policy:
+         ("PREALLOCATED", "PREALLOCATED_WITH_REALLOC", "DYNAMIC", "DYNAMIC_REUSABLE")-->
+    <xs:simpleType name="historyMemoryPolicyType">
         <xs:restriction base="xs:string">
-            <xs:pattern value="((1?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])\.){3}(1?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])" />
+            <xs:enumeration value="PREALLOCATED"/>
+            <xs:enumeration value="PREALLOCATED_WITH_REALLOC"/>
+            <xs:enumeration value="DYNAMIC"/>
+            <xs:enumeration value="DYNAMIC_REUSABLE"/>
         </xs:restriction>
     </xs:simpleType>
 
-    <xs:simpleType name="ipv6AddressType">
-        <xs:restriction base="xs:string">
-            <xs:pattern value="::[A-Fa-f0-9]{4}|([A-Fa-f0-9]{1,4}:){7}[A-Fa-f0-9]{1,4}" />
-        </xs:restriction>
-    </xs:simpleType>
+    <!--Allocation Type:
+        ├ initial   [uint32],
+        ├ maximum   [uint32],
+        └ increment [uint32] -->
+    <xs:complexType name="allocationConfigType">
+        <xs:all minOccurs="0">
+            <xs:element name="initial" type="uint32" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="maximum" type="uint32" default="0" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="increment" type="uint32" minOccurs="0" maxOccurs="1"/>
+        </xs:all>
+    </xs:complexType>
 
+
+
+
+    <!--| LOCATOR TYPES DEFINITION |-->
+    <!--udpv4 Locator:
+        ├ port    [uint32],
+        └ address [ipv4Address]-->
     <xs:complexType name="udpv4LocatorType">
-        <xs:all minOccurs="0">
-            <xs:element name="port" type="uint32Type" minOccurs="0"/>
-            <xs:element name="address" type="stringType" minOccurs="0"/>
+        <xs:all>
+            <xs:element name="port" type="uint32" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="address" type="ipv4Address" minOccurs="0" maxOccurs="1"/>
```

Comment:
I have tried erasing the value inside a "address" tag, and the validator says it is valid.

Reply:
Please, make sure this does not happen with the rest of the elements that need a value inside.